src/uct/base/uct_iface.c

```diff
@@ -62,12 +63,19 @@ static ucs_status_t uct_iface_stub_am_handler(void *arg, void *data,
     const size_t dump_len = 64;
     uint8_t id            = (uintptr_t)arg;
     char dump_str[(dump_len * 4) + 1]; /* 1234:5678\n\0 */
+    char *bt_str;
+    size_t bt_size;
+    FILE *bt;
 
     ucs_warn("got active message id %d, but no handler installed", id);
-    ucs_warn("payload:\n%s", ucs_str_dump_hex(data,
-                                              ucs_min(length, dump_len),
-                                              dump_str, sizeof(dump_str),
-                                              16));
+    ucs_warn("payload %zu of %zu bytes:\n%s", ucs_min(length, dump_len), length,
+             ucs_str_dump_hex(data, ucs_min(length, dump_len),
+                              dump_str, sizeof(dump_str), 16));
+    bt = open_memstream(&bt_str, &bt_size);
```

Comment:
not sure it's a good idea, log string has a limited size and backtrace can be long
also need to check for NULL

Reply:
hmmm

usually unhandled message cause hang or crash and logs should not be overloaded

Reply:
added NULL evaluation

Reply:
can you check if a big backtrace is really printed in this case? and attach the output?

Reply:
added for 20 strings limit to prevent log overloading...
output, it seems backtrace is limited by some value:
```
[1568881800.354417] [hpchead:11782:0]         ucp_am.c:561  UCX  WARN  payload 64 of 72 bytes:
[1568881800.354417] [hpchead:11782:0]         ucp_am.c:561  UCX  WARN  40000000:00000000:40404040:40404040
[1568881800.354417] [hpchead:11782:0]         ucp_am.c:561  UCX  WARN  40404040:40404040:40404040:40404040
[1568881800.354417] [hpchead:11782:0]         ucp_am.c:561  UCX  WARN  40404040:40404040:40404040:40404040
[1568881800.354417] [hpchead:11782:0]         ucp_am.c:561  UCX  WARN  40404040:40404040:40404040:40404040
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN  ==== backtrace (tid:  11782) ====
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   0 0x0000000000056080 ucs_debug_print_backtrace()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/ucs/debug/debug.c:445
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   1 0x000000000001d410 ucp_am_handler()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/ucp/core/ucp_am.c:567
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   2 0x000000000001521f uct_iface_invoke_am()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/uct/base/uct_iface.h:628
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   3 0x000000000001594c uct_mm_iface_invoke_am()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/uct/sm/mm/base/mm_iface.h:151
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   4 0x0000000000015df8 uct_mm_iface_poll_fifo()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/uct/sm/mm/base/mm_iface.c:254
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   5 0x0000000000015f6c uct_mm_iface_progress()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/uct/sm/mm/base/mm_iface.c:278
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   6 0x000000000002ede4 ucs_callbackq_dispatch()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/ucs/datastruct/callbackq.h:211
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   7 0x0000000000035511 uct_worker_progress()  /auto/mtrswgwork/sergeyo/vs-projects/ucx/src/uct/api/uct.h:2203
[1568881801.119404] [hpchead:11782:0]         ucp_am.c:581  UCX  WARN   8 0x000000000079484f ucp_test_base::entity::pro
```

Reply:
@hoopoepg we need a method to print the backtrace without any size limit
so i would add function ucs_log_print_backtrace() which would print it to the configured log file (which could be stdout)
or we can print each line in a separate log message

Reply:
@yosefe added macro ucs_for_each_line which allows to process long string by lines

Reply:
i think the approach in this PR should be different: instead: of saving everything to a memory buffer, add method to print a debug backtrace to log. For example: ucs_debug_log_backtrace(log_level_t, int strip)
In order to reduce code duplication with current function, and also between HAVE_DETAILED_BACKTRACE=0/1, i suggest to improve "ucs_debug_backtrace_create" and "backtrace_next" iterator, so it will fallback internally to glibc backtrace/backtrace_symbols if HAVE_DETAILED_BACKTRACE=0, and backtrace_next to check ucs_debug_backtrace_is_excluded internally.

Reply:
yep, remember about this... will complete it soon :)

Reply:
@yosefe re-implemented